internal/manifests/compactor.go

```diff
@@ -97,7 +97,8 @@ func NewCompactorStatefulSet(opt Options) *appsv1.StatefulSet {
 		},
 	}
 
-	compactorLabels := ComponentLabels("compactor", opt.Name)
+	l := ComponentLabels("compactor", opt.Name)
+	a := commonAnnotations(opt.ConfigSHA1)
```

Comment:
does this mean that for every component, we would have the same hash?
i thought we had discussed the idea that each component would have a hash based on it's specific configurations

Reply:
This would be a next step to split the configuration per component. Right now some configuration is shared among components (e.g. `storage_config` and `limits_config`). I suggest to flesh a per-component+shared hashing in a follow-up PR.